docs/src/references/c8y-configuration-management.md

```diff
@@ -201,71 +201,111 @@ To provide/consume configuration files to/from external devices, the HTTP filetr
-    * on an incoming PUT request `tedge_agent` stores the contained file to `/http-root/<path>`
-    * on an incoming GET request `tedge_agent` sends the file `/http-root/<path>` to the requester (i.E. the external device)
-    * `tedge_agent` does not force any predefined folder structure within `/http-root/`. However for configuration management it is best practice to use folder structure as below:
-      * `/http-root/config/current/<child-id>/<config filename>`<br/>
-         for config-files provided by the external device to the thin-edge device (current state - or latest known state)<br/>
-      * `/http-root/config/desired/<child-id>/<config filename>`<br/>
-        for config-files to be consumed by the external device (desired state)
-  * `tedge_agent` notifies over MQTT any change of a file under `/http-root/` (e.g. when uploaded by a external-device or updated by a local process). The notification messages are published on the topic `tedge/filetransfer_change/{path}`.<br/>
-    TO-BE-DEFINED: Content of payload?
-      
-      Example:
-      * HTTP PUT URL: `http://<thin-edge IP address>/tedge/filetransfer/config/current/child1/bar.conf`
-      * relating MQTT notification topic: `/tedge/filetransfer_change/config/current/child1/bar.conf`
-
-  * `c8y_configuration_plugin` allows to handle two versions for each configuration file (_current_ and _desired_).
-    * The configuration file's record `path` is treated as path of the _current_ version of a configuration file (see to section 'Configuration' above).
-    * The configuration file's record `desired` can be used to define the path of the _desired_ version of a configuration file. If `desired` is not defined, then `path` is treated as _desired_ version also.
-    * On c8y read request for a configuration, `c8y_configuration_plugin` returns the `current` file content.
-    * On c8y write request for a configuration, `c8y_configuration_plugin` updates the `desired` file content and waits for the `current` content to be updated to return the operation status to cumulocity.  
-    * The current behavior of c8y_configuration_plugin can be simulated with desired = current = path when only a path is provided.
-
-Example Plugin Config:
-```shell
-$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
-files = [
-    { type = 'bar.conf', child_id = 'child1', current = '/http-root/config/current/child1/bar.conf', desired = '/http-root/config/desired/child1/bar.conf' } }
-  ]
-```
+  * `tedge_agent` serves PUT and GET requests for temporary file up/downloads
+    * a temporary file can be uploaded with an HTTP PUT request to `http://<thin-edge IP address>/tedge/tmpfiles`,
+      where the HTTP response contains a _temporary URL_.
+    * a temporary file can be downloaded with an HTTP GET request to `http://<ip address of thin-edge device>/<temporary URL>`
+    * a local API allows the plugin to access (expose and obtains) a temporary file directly via the thin-edge device's local filesystem
```

Comment:
Thats true, technically the plugin could use the REST API, and download/upload files via HTTP GET and PUT. However I described here a "local API" that uses the local filesystem between plugin and filetransfer, to achieve both things below:
   1) avoid that there are two copies of the same file at any point in time
   2) avoid that the file content is transfered/copied from one to another place, when the plugin obtains/exposes the file

Reason for that is to safe disc space and CPU use, especially for large files. Maybe we can postpone that for config-files, since these might usually really small. However when we start with SW Management or Firmware Management we definitely need to provide somehting in that direction.

Reply:
> Thats true, technically the plugin could use the REST API, and download/upload files via HTTP GET and PUT. However I described here a "local API" that uses the local filesystem between plugin and filetransfer, to achieve both things below [...]

However these things are not needed for configuration files, since these are quite small compares to files for SW Management or Firmware Management. So the C8Y Config Plugin can use the REST API. We shall consider such kind of local API later, when we adopt the filetransfer to SW Management or Firmware Management.